internal/component/loki/source/docker/internal/dockertarget/target.go

```diff
@@ -321,3 +313,68 @@ func (t *Target) getStreamLabels(logStream string) model.LabelSet {
 
 	return filtered
 }
+
+// dockerChunkWriter implements io.Writer to preprocess and reassemble Docker log frames.
+type dockerChunkWriter struct {
+	writer      io.Writer
+	logger      log.Logger
+	buffer      *bytes.Buffer
+	isBuffering bool
+}
+
+var bufferPool = sync.Pool{
+	New: func() interface{} {
+		buf := &bytes.Buffer{}
+		buf.Grow(dockerMaxChunkSize * 2)
+		return buf
```

Comment:
May as well initialize with that size instead of Growing.

```suggestion
	return bytes.NewBuffer(make([]byte, 0, dockerMaxChunkSize * 2))
```